### ISSUE TITLE
Sentence.text() adds extra whitespace

### ISSUE THREAD

**Issue by econnelly-amplify** [spencermountain/nlp_compromise#74]
At least in one example, splitting a text into POS Sentences adds extra whitespace in the sentence.text() function. (See two spaces between "we've" and "known" in the returned value)

Example:
```
sentence = nlp.pos("This is different than TV series as we've known them.").sentences[0]
sentence.text()
This is different than TV series as we've  known them.
```

**Comment by econnelly-amplify** [comment]
Btw, I plan to take a look into this, but just in case I can't resolve it I wanted to document the example in a bug case for you. Thanks for the project!

**Comment by spencermountain** [comment]
hey! thanks.
i'm *days* away from releasing v2, which has a more intellegent text() method, wanna give it a go?
https://github.com/spencermountain/nlp_compromise/tree/2.0

**Comment by econnelly-amplify** [comment]
Oh cool, sure! I'll try it and see what happens, thanks!

**Comment by econnelly-amplify** [comment]
Well it seems to have fixed the extra whitespace issue (cool!). But now I just have to figure out how to maintain the contraction.  It seems like nlp keeps replacing it with the two-word equivalent.
```
nlp.text("This is a story we've heard.").sentences[0].text()
This is a story we have heard.
```
Still, I guess we can close the issue now. Thanks!

**Comment by spencermountain** [comment]
what would be the best behaviour for contractions? i agree that .text() should return the pure form, but internally the contraction needs to be understood as two parts of speech.
what's your use-case, if you don't mind me asking?

**Comment by econnelly-amplify** [comment]
Hi Spencer, I don't mind at all.

I saw some comment about maybe having a .raw_text() which might work.

My use case is not too complicated really. I have an inputText of a bunch
of sentences, and I want to know the character startIndex of each sentence
(I plan to reprint the whole inputText while highlighting certain words in
a red font, and right now I am using character index to know which words to
highlight).  So I tried using inputText.indexOf(sentence.text()) to find
the character start Index.  And that's when I saw the text was slightly
different, because .indexOf() returned -1.

On Sun, Dec 20, 2015 at 2:22 PM, spencer kelly <notifications@github.com>
wrote:

> what would be the best behaviour for contractions? i agree that .text()
> should return the pure form, but internally the contraction needs to be
> understood as two parts of speech.
> what's your use-case, if you don't mind me asking?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spencermountain/nlp_compromise/issues/74#issuecomment-166148441>
> .
>



-- 
Eric Connelly
econnelly@amplify.com
